clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala

```diff
@@ -73,19 +76,26 @@ abstract class PersistenceDAO(recoveryParallelism: Int) extends LazyLogging {
     * @param index    Typeclass providing information on where to persist the
     *                 metadata update.
     */
-  def writeUpdate[D <: ClioDocument](document: D)(
+  def writeUpdate[D <: ClioDocument](
+    document: D,
+    dt: OffsetDateTime = OffsetDateTime.now()
+  )(
     implicit ec: ExecutionContext,
     index: ElasticsearchIndex[D]
-  ): Future[Unit] =
-    Future {
-      val writePath =
-        Files.createDirectories(rootPath.resolve(index.currentPersistenceDir))
-      val written = Files.write(
-        writePath.resolve(s"${document.persistenceFilename}"),
-        index.indexable.json(document).getBytes
-      )
-      logger.debug(s"Wrote document $document to ${written.toUri}")
-    }
+  ): Future[Unit] = Future {
+    import index.encoder
+
+    val jsonString = ModelAutoDerivation.defaultPrinter.pretty(document.asJson)
+
+    val writePath =
+      Files.createDirectories(rootPath.resolve(index.persistenceDirForDatetime(dt)))
+
+    val written = Files.write(
+      writePath.resolve(ClioDocument.persistenceFilename(document.upsertId)),
+      jsonString.getBytes
+    )
+    logger.debug(s"Wrote document $document to ${written.toUri}")
```

Comment:
Not sure what `toUri()` buys you here? Is it a shorter way to ensure you print the absolute path than writing `toAbsolutePath()`?

Reply:
`toUri()` includes the `gs://` prefix in its stringified form, and (I'm pretty sure) `toAbsolutePath` doesn't.